kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/FaqAlRichMessage.java

```diff
@@ -0,0 +1,98 @@
+    }
+
+    @Override
+    protected void setupAlRichMessage(ViewGroup faqLayout, ViewGroup faqReplyLayout, ALRichMessageModel model) {
+        super.setupAlRichMessage(faqLayout, faqReplyLayout, model);
+        if(model != null) {
+            TextView headerText = faqLayout.findViewById(R.id.headerText);
+            TextView titleText = faqLayout.findViewById(R.id.questionText);
+            TextView descriptionText = faqLayout.findViewById(R.id.bodyText);
+            TextView buttonLabel = faqReplyLayout.findViewById(R.id.buttonLabel);
+            TextView actionYes = faqReplyLayout.findViewById(R.id.actionYes);
+            TextView actionNo = faqReplyLayout.findViewById(R.id.actionNo);
+
+            if (model.getPayload() != null) {
+                ALRichMessageModel.ALPayloadModel payload = (ALRichMessageModel.ALPayloadModel) GsonUtils.getObjectFromJson(model.getPayload(), ALRichMessageModel.ALPayloadModel.class);
+                if (payload != null) {
+                    if (!TextUtils.isEmpty(payload.getTitle())) {
+                        titleText.setVisibility(View.VISIBLE);
+                        titleText.setText(getHtmlText(payload.getTitle()));
+                    } else {
+                        titleText.setVisibility(View.GONE);
+                    }
+
+                    if (!TextUtils.isEmpty(payload.getDescription())) {
+                        descriptionText.setVisibility(View.VISIBLE);
+                        descriptionText.setText(getHtmlText(payload.getDescription()));
+                    } else {
+                        descriptionText.setVisibility(View.GONE);
+                    }
+
+                    List<ALRichMessageModel.AlButtonModel> actionModel = payload.getButtons();
+                    if (actionModel != null) {
+                        faqReplyLayout.setVisibility(View.VISIBLE);
+
+                        if (!TextUtils.isEmpty(payload.getButtonLabel())) {
+                            buttonLabel.setVisibility(View.VISIBLE);
+                            buttonLabel.setText(payload.getButtonLabel());
+                        } else {
+                            buttonLabel.setVisibility(View.GONE);
+                        }
+
+                        if (actionModel.size() > 0 && actionModel.get(0) != null) {
+                            if (!TextUtils.isEmpty(actionModel.get(0).getName())) {
+                                actionYes.setVisibility(View.VISIBLE);
+                                actionYes.setText(actionModel.get(0).getName());
+                                setActionListener(actionYes, model, actionModel.get(0), payload);
+                            } else {
+                                actionYes.setVisibility(View.GONE);
+                            }
+                        }
+
+                        if (actionModel.size() > 1 && actionModel.get(1) != null) {
+                            if (!TextUtils.isEmpty(actionModel.get(1).getName())) {
```

Comment:
looks to be duplicate code same as line no. 73 to 78, please move it to a common code

Reply:
done